d2bs/kolbot/libs/OOG.js

```diff
@@ -668,35 +662,31 @@ const ControlAction = {
 			throw new Error("clickRealm: Invalid realm!");
 		}
 
-		let control, currentRealm;
+		let currentRealm;
 
 		me.blockMouse = true;
 
 		MainLoop:
 		while (true) {
 			switch (getLocation()) {
 			case sdk.game.locations.MainMenu:
-				control = getControl(6, 264, 391, 272, 25);
+				switch (Control_1.Gateway.text.split(getLocaleString(11049).substring(0, getLocaleString(11049).length - 2))[1]) {
```

Comment:
dang you are fast :D.

But you have a problem here. if `Control_1.Gateway.text` is null you will get some ugly `null has no method split` and crash. So I guess you cannot get around checking the output of the wrapper unless you make tests within the wrapper itself (assuming you will always want that localeString stuff for texts) and catch any problems there.
Then you should still have a default case for the error even if you don't handle it imho.

Reply:
I changed it to break from the switch if it fails and added a retry amount to prevent infinite loop.